src/Compilers/Core/Portable/SourceGeneration/UserFunction.cs

```diff
@@ -52,9 +50,7 @@ internal static Action<TInput> WrapUserAction<TInput>(this Action<TInput> userAc
                 {
                     userAction(input);
                 }
-#pragma warning disable CS0618 // ReportIfNonFatalAndCatchUnlessCanceled is obsolete; tracked by https://github.com/dotnet/roslyn/issues/58375
-                catch (Exception e) when (FatalError.ReportIfNonFatalAndCatchUnlessCanceled(e))
-#pragma warning restore CS0618 // ReportIfNonFatalAndCatchUnlessCanceled is obsolete
+                catch (Exception e) when (ExceptionUtilities.IsNotCancelled(e))
```

Comment:
It seems a bit odd we'd be filtering cancellation exceptions here if we didn't have a cancellation token to start with. Should we be getting rid of this overload entirely? Or should we actually be catching these since the generator shouldn't be throwing exceptions in the first place?

Reply:
(feel free to fix this in a later PR, since this is otherwise a mechanical refactoring)

Reply:
Yeah, @jasonmalinowski is probably right here. Given that we *cant* cancel in this scenario, we probably shouldn't be filtering it at all.